src/main/java/org/reaktivity/nukleus/mqtt/internal/stream/MqttServerFactory.java

```diff
@@ -1204,6 +1307,39 @@ private void onDecodeConnect(
                 reasonCode = PROTOCOL_ERROR;
             }
 
+            final MqttPropertiesFW properties = packet.properties();
+
+            final OctetsFW propertiesValue = properties.value();
+            final DirectBuffer decodeBuffer = propertiesValue.buffer();
+            final int decodeOffset = propertiesValue.offset();
+            final int decodeLimit = propertiesValue.limit();
+
+            boolean topicAliasMaximumSet = false;
```

Comment:
Please track `topicAliasMaximum` with an initially illegal value to detect when it is set, and then remove this separate boolean tracking variable which is no longer needed.